### PR TITLE
Add ByteStringSerializer

### PR DESCRIPTION
**Pull request by andrewsmartin**
This one is for the failing `BigTableExampleTest` in beam-master.

### CODE REVIEW

File: scio-core/src/main/scala/com/spotify/scio/coders/ByteStringSerializer.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+package com.spotify.scio.coders
+
+import com.esotericsoftware.kryo.Kryo
+import com.esotericsoftware.kryo.io.{Input, Output}
+import com.google.protobuf.ByteString
+import com.twitter.chill.KSerializer
+
+class ByteStringSerializer extends KSerializer[ByteString] {
+  override def read(kryo: Kryo, input: Input, tpe: Class[ByteString]): ByteString =
+    ByteString.copyFromUtf8(input.readString)
```
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/coders/ByteStringSerializer.scala]: You shouldn't use string here. `ByteString` can contain `Array[Byte]`s that are not legit strings, e.g. weird encoding, `\0` in the middle.

Comment (nevillelyh) [reply scio-core/src/main/scala/com/spotify/scio/coders/ByteStringSerializer.scala]: Also what makes `ByteString` not deterministic out of the box? It has fields other than the `Array[Byte]`?

Comment (nevillelyh) [reply scio-core/src/main/scala/com/spotify/scio/coders/ByteStringSerializer.scala]: Hm... so `ByteString` is an abstract class and some impls like `BoundedByteString` has extra stateful fields like offset and length.

Comment (andrewsmartin) [reply scio-core/src/main/scala/com/spotify/scio/coders/ByteStringSerializer.scala]: That makes sense, I'll change this to read/write `Array[Byte]` instead.

**Comment by nevillelyh** [comment]
Cherry picked into the branch.

**Pull request closed without merging**